jgrapht-core/src/main/java/org/jgrapht/alg/isomorphism/RefinementAbstractIsomorphismInspector.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * (C) Copyright 2018-2018, by Christoph Grüne and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.isomorphism;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphMapping;
+import org.jgrapht.GraphType;
+
+import java.util.Iterator;
+
+/**
+ * Base implementation of the color refinement algorithms using its feature of detecting
+ * <a href="http://mathworld.wolfram.com/GraphIsomorphism.html">isomorphism between two graphs</a>
+ * as described in
+ * C. Berkholz, P. Bonsma, and M. Grohe.  Tight lower and upper bounds for the complexity of canonical
+ * colour refinement. Theory of Computing Systems,doi:10.1007/s00224-016-9686-0, 2016 (color refinement)
+ * and
+ * reference will follow (individualization refinement).
+ *
+ * @param <V> the vertex type
+ * @param <E> the edge type
+ *
+ * @author Christoph Grüne
+ */
+public abstract class RefinementAbstractIsomorphismInspector<V, E> implements IsomorphismInspector<V, E> {
```

Comment:
There is no need to use an abstract class here, as you are not reusing any code. Delete this class completely and move the constructor and the two graph references inside your `ColorRefinementIsomorphismInspector` class.

Reply:
Later, we will use this class for our individualization refinement algorithms. Should we delete it anyways?

Reply:
We do refactoring the other way around. When we have two classes which implement the same interface and both use common methods, then we refactor using "extract superclass". 

If there is a need later on, you can then use an abstract superclass.